app/models/concerns/tag_list.rb

```diff
@@ -10,18 +10,23 @@ def tag_list=(input_tag_names)
     is_course = self.kind_of?(Course)
     object = self
     # we need to dereference or delete tags in a way that updating tags does not effect other courses or lessons
-    if self.tags.any?
-      self.tags.each do |tag|
-	key_word = is_course ? KeyWord.find_by(course_id: object.id, tag_id: tag.id) : KeyWord.find_by(lesson_id: object.id, tag_id: tag.id)
-        # see if tag belogs to any other courses or lessons
-        # if yes - delete association, if not delete the tag
-        (tag.lessons.count + tag.courses.count) > 1 ? key_word.delete : tag.delete
-      end
-    end
+    # if self.tags.any?
+    #   self.tags.each do |tag|
+	  #     key_word = is_course ? KeyWord.find_by(course_id: object.id, tag_id: tag.id) : KeyWord.find_by(lesson_id: object.id, tag_id: tag.id)
+    #     # see if tag belogs to any other courses or lessons
+    #     # if yes - delete association, if not delete the tag  
+    #     if (tag.lessons.count + tag.courses.count) > 1 
+    #       key_word.destroy 
+    #     else 
+    #       tag.destroy
+    #     end 
+    #   end
+    # end
     # create tags and/or associations through key_words for all values passed in
     self.reload
     tag_names = input_tag_names.collect{|t| t.strip.downcase}.uniq
     tag_names = tag_names.reject {|x| x == ""}
+    self.key_words.destroy_all
```

Comment:
I think this is fine but let's make a mental note that we could probably work this in above somewhere in lines 13-20 to avoid redundancy.